admin/module/catalog-admin-module-sources/src/main/webapp/js/view/ConfigurationEdit.view.js

```diff
@@ -121,7 +121,7 @@ define([
             _.each(this.collectionArray.models, function(model) {
                 values.push(model.get('value'));
             });
-            this.configuration.get('properties').set(this.model.get('id'), values);
+            this.configuration.get('properties').set(this.model.get('id'), values.length === 0 ? null : values);
```

Comment:
Inside of ConfigurationAdmin.java the updateForLocation method loops through all the attributes and sets them to an empty string if they're null. If an array is empty, it won't get set to null and then will cause the configuration service to throw an invalid argument exception.